aiida_common_workflows/cli/launch.py

```diff
@@ -103,3 +105,102 @@ def cmd_relax(
+    """Relax a crystal structure using the common relax workflow for one of the existing plugin implementations.
+
+    The command will automatically try to find and load the codes that are required by the plugin workflow. If no code
+    is installed for at least one of the calculation engines, the command will fail. Use the `--show-engine` flag to
+    display the required calculation engines for the selected plugin workflow.
+    """
+    # pylint: disable=too-many-locals
+    from aiida.orm import QueryBuilder, Code, Dict
+    from aiida_common_workflows.plugins import get_entry_point_name_from_class
+    from aiida_common_workflows.workflows.eos import EquationOfStateWorkChain
+    from aiida_common_workflows.workflows.relax import RelaxType
+
+    process_class = load_workflow_entry_point('relax', plugin)
+    generator = process_class.get_inputs_generator()
+
+    if not generator.is_valid_protocol(protocol):
+        protocols = generator.get_protocol_names()
+        process_class_name = process_class.__name__
+        message = f'`{protocol}` is not implemented by `{process_class_name}` workflow: choose one of {protocols}'
+        raise click.BadParameter(message, param_hint='protocol')
+
+    if show_engines:
+        for engine in generator.get_calc_types():
+            schema = generator.get_calc_type_schema(engine)
+            click.secho(engine, fg='red', bold=True)
+            click.echo('Required code plugin: {}'.format(schema['code_plugin']))
+            click.echo('Engine description:   {}'.format(schema['description']))
+
+        return
+
+    engines = {}
+
+    for engine in generator.get_calc_types():
+        schema = generator.get_calc_type_schema(engine)
+        engines[engine] = {
+            'options': {
+                'resources': {
+                    'num_machines': 1
+                },
+                'max_wallclock_seconds': 86400,
+            }
+        }
+        code_plugin = schema['code_plugin']
+        query = QueryBuilder().append(Code, filters={'attributes.input_plugin': code_plugin})
+
+        code = query.first()
+
+        if code is None:
+            raise click.UsageError(f'could not find a configured code for the plugin `{code_plugin}`.')
+
+        engines[engine]['code'] = code[0].full_label
+
+    inputs = {
+        'structure':
+        structure,
+        'generator_inputs':
+        Dict(
+            dict={
+                'calc_engines': engines,
+                'protocol': protocol,
+                'relaxation_type': RelaxType.ATOMS.value,
+                'threshold_forces': threshold_forces,
+                'threshold_stress': threshold_stress
+            }
+        ),
+        'sub_process_class':
+        get_entry_point_name_from_class(process_class).name,
+    }
+    utils.launch_process(EquationOfStateWorkChain, daemon, **inputs)
+
+
+@cmd_launch.command('plot-eos')
+@options_core.NODE()
```

Comment:
Shouldn't be a `@click.argument` instead of an `@option_core`

Reply:
Yes, I agree an argument makes more sense here